### PR TITLE
Add ionCube to the list of incompatible extensions + add INI setting to ignore incompatible extensions

### PR DESCRIPTION
**Pull request by iamluc**
### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: ext/excluded_modules.c [hunk 1]
```diff
@@ -6,16 +6,23 @@
 #include <ext/standard/php_versioning.h>
 
 #include <components/log/log.h>
+#include "configuration.h"
 
 bool ddtrace_is_excluded_module(zend_module_entry *module, char *error) {
+    if (strcmp("ionCube Loader", module->name) == 0) {
```
Comment (bwoebi) [inline ext/excluded_modules.c:12]: `#if PHP_VERSION_ID >= 80000` as far as I know. Haven't heard about incompatibilities on PHP 7.

Comment (bwoebi) [inline ext/excluded_modules.c:12]: `#if PHP_VERSION_ID >= 80000` - on PHP 7 it's working to my knowledge - it's not crashing, but instrumentation of the encrypted code itself will not trigger. But that can be fine, as often e.g. framework code isn't encrypted, just the app code.

Comment (iamluc) [reply ext/excluded_modules.c:13]: Updated

File: ext/excluded_modules.c [hunk 2]
```diff
@@ -50,18 +58,27 @@ void ddtrace_excluded_modules_startup() {
     zend_module_entry *module;
 
     ddtrace_has_excluded_module = false;
+    bool inject_force = (IS_TRUE == Z_TYPE(zai_config_memoized_entries[DDTRACE_CONFIG_DD_INJECT_FORCE].decoded_value));
```
Comment (bwoebi) [inline ext/excluded_modules.c:61]:
```suggestion
    bool inject_force = get_global_DDTRACE_CONFIG_DD_INJECT_FORCE();
```

File: ext/configuration.h [hunk 3]
```diff
@@ -226,6 +226,7 @@ enum ddtrace_sampling_rules_format {
     CONFIG(INT, DD_OPENAI_SPAN_CHAR_LIMIT, "128")                                                              \
     CONFIG(DOUBLE, DD_OPENAI_SPAN_PROMPT_COMPLETION_SAMPLE_RATE, "1.0")                                        \
     CONFIG(DOUBLE, DD_OPENAI_LOG_PROMPT_COMPLETION_SAMPLE_RATE, "0.1")                                         \
+    CONFIG(BOOL, DD_INJECT_FORCE, "false")                                                                     \
```
Comment (bwoebi) [inline ext/configuration.h:229]:
```suggestion
    CONFIG(BOOL, DD_INJECT_FORCE, "false", .ini_change = zai_config_system_ini_change)                        \
```

Comment (bwoebi) [inline ext/configuration.h:229]:
```suggestion
    CONFIG(BOOL, DD_INJECT_FORCE, "false", .ini_change = zai_config_system_ini_change)                             \
```

**Review by bwoebi — APPROVED** [review]
Looks good with force_inject.

**Pull request merged**